src/NUnitConsole/nunit3-console.tests/TestEventHandlerTests.cs

```diff
@@ -71,6 +76,83 @@ public void MultipleEvents(string[] reports, string labels, string expected)
             Assert.That(Output, Is.EqualTo(expected));
         }
 
+        [TestCaseSource(nameof(ErrorData))]
+        public void ErrorOutputGoesToOutputWhenNotOverridden(string labels, string expected)
+        {
+            var handler = new TestEventHandler(_writer, _writer, labels);
+
+            handler.OnTestEvent(ErrorOutputXml);
+
+            Assert.That(Output, Is.EqualTo(expected));
+        }
+
+        [TestCaseSource(nameof(ErrorData))]
+        public void ErrorOutputGoesToErrorOutputWhenOverridden(string labels, string expected)
+        {
+            var handler = new TestEventHandler(_writer, _errWriter, labels);
+
+            handler.OnTestEvent(ErrorOutputXml);
+
+            Assert.That(ErrorOutput, Is.EqualTo(expected));
+            Assert.That(Output, Does.Not.Contain("This is an error"));
+        }
+
+        [Test]
+        public void ErrorWriteDoesNotIncludeNewlinesInSameTest()
+        {
+            var handler = new TestEventHandler(_writer, _errWriter, "Before");
+
+            handler.OnTestEvent(string.Format(ErrorOutputFormat, "ErrorTest", "One"));
+            handler.OnTestEvent(string.Format(ErrorOutputFormat, "ErrorTest", "Two"));
+
+            Assert.That(ErrorOutput, Is.EqualTo($"=> ErrorTest{Environment.NewLine}OneTwo"));
+        }
+
+        [Test]
+        public void ErrorWriteLineDoesIncludeNewlinesInSameTest()
+        {
+            var handler = new TestEventHandler(_writer, _errWriter, "Before");
+
+            handler.OnTestEvent(string.Format(ErrorOutputFormat, "ErrorTest", "One\r\n"));
+            handler.OnTestEvent(string.Format(ErrorOutputFormat, "ErrorTest", "Two\r\n"));
```

Comment:
Why are these two explicit `\r\n`s - and everything else uses Environment.NewLine? Is that to do with what the framework writes?

Reply:
I'm glad this stuff's explicitly tested thought - there's some subtle functionality here. 😄